qiskit/circuit/library/n_local/n_local.py

```diff
@@ -306,6 +306,86 @@ class NLocal(BlueprintCircuit):
 
         The :func:`.n_local` function constructs a functionally equivalent circuit, but faster.
 
+    Examples:
+
+        Minimal usage::
+
+            from qiskit.circuit.library import NLocal
+            from qiskit.circuit.library.standard_gates import RYGate, CXGate
+
+            nlocal = NLocal(
+                num_qubits=3,
+                rotation_blocks=RYGate(1),
+                entanglement_blocks=CXGate(),
+                entanglement='linear',
+                reps=2,
+                insert_barriers=True
+            )
+            nlocal.decompose().draw('mpl')
+
+        Multiple rotation and entanglement blocks::
+
+            from qiskit.circuit.library.standard_gates import RYGate, RZGate, CXGate, CZGate
+
+            nlocal = NLocal(
+                num_qubits=4,
+                rotation_blocks=[RYGate(1), RZGate(1)],
```

Comment:
Could you give at least one of them a free parameter? That's typically the use case for nlocal circuits 🙂